tests/lib/nested.sh

```diff
@@ -51,6 +51,41 @@ nested_wait_for_no_ssh() {
     done
 }
 
+
+nested_wait_vm_ready() {
+    local retry=${1:-120}
+    local log_limit=${2:-60}
+
+    local output_lines=0
+    local serial_log="$NESTED_LOGS_DIR"/serial.log
+    while true; do
+        retry=$(( retry - 1 ))
+
+        # Check the timeout is reached
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for vm ready. Aborting!"
+            return 1
+        fi
+
+        # Check the vm is active
+        if ! systemctl is-active "$NESTED_VM"; then
+            echo "Unit $nested_unit is not active. Aborting!"
+            return 1
+        fi
+
+        # Check during $limit seconds that the serial log is growing
+        retry -n "$log_limit" --wait 1 test "$(wc -l "$serial_log" | awk '{print $1;}')" -gt "$output_lines"
```

Comment:
The left hand side of the `test ... -gt ...` will be evaluated before being passed to `retry`.

Also `| awk '{print $1;}'` seems useless to me.

Reply:
@valentindavid thanks for the review
about the awk '{print $1;}' it is because wc prints the lines and the file name.
> wc -l NEWS.md 
66 NEWS.md

Reply:
What about `wc -l <"$serial_log"`?